### PR TITLE
Fix ISO provisioning with kickstart

### PR DESCRIPTION
**Pull request by borod108**
Provisioning with kickstart was not implemented for version 4 of the api.

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1613326

### CODE REVIEW

**Comment by borod108** [comment]
@masayag @pkliczewski it is wiped because I want to add a spec, I did verify it by patching an environment, please review.

File: spec/models/manageiq/providers/redhat/infra_manager/provision/configuration_spec.rb [hunk 1]
```diff
@@ -13,13 +13,43 @@
...
+    context "with ovirtsdk4" do
+      let(:vm_proxy)        { OvirtSDK4::Vm.new }
+      let(:vm_service)      { double("OvirtSDK4::VmService", :get => vm_proxy) }
+      let(:connection)      { double("OvirtSDK4::Connection") }
+      let(:ovirt_services)  { double("ManageIQ::Providers::Redhat::InfraManager::OvirtServices::Strategies::V4") }
+      let(:provider_object) { ManageIQ::Providers::Redhat::InfraManager::OvirtServices::Strategies::V4::VmProxyDecorator.new(vm_service, connection, ovirt_services) }
+
+      it "updates the vm with the right payload" do
+        task.options[:customization_template_id] = cust_template.id
+        #allow(vm_service).to receive(:get).and_return()
```
Comment (pkliczewski) [inline spec/models/manageiq/providers/redhat/infra_manager/provision/configuration_spec.rb]: Why do we need this commented line?

Comment (borod108) [reply spec/models/manageiq/providers/redhat/infra_manager/provision/configuration_spec.rb]: what commented line ,which I totally did not forget to remove, do you mean? ;)

Comment (pkliczewski) [reply spec/models/manageiq/providers/redhat/infra_manager/provision/configuration_spec.rb]: Line 36 `#allow(vm_service).to receive(:get).and_return()`

Comment (borod108) [reply spec/models/manageiq/providers/redhat/infra_manager/provision/configuration_spec.rb]: @pkliczewski done :)

**Comment by borod108** [comment]
@pkliczewski @masayag merge?

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit 8a8e85a0be27f45707a873deb1fe828ea2a91682
Author: Piotr Kliczewski <piotr.kliczewski@gmail.com>
Date:   Mon Oct 8 16:13:46 2018 +0200

    Merge pull request #296 from borod108/bugs/1613326iso_prov
    
    Fix ISO provisioning with kickstart
    
    (cherry picked from commit 9d27961b57a542a3259984fdd436d0a25c187435)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1613326
```

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit dadbea59d0bf5cd473051548ddda12a4195a695e
Author: Piotr Kliczewski <piotr.kliczewski@gmail.com>
Date:   Mon Oct 8 16:13:46 2018 +0200

    Merge pull request #296 from borod108/bugs/1613326iso_prov
    
    Fix ISO provisioning with kickstart
    
    (cherry picked from commit 9d27961b57a542a3259984fdd436d0a25c187435)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1637185
```

**Pull request merged**